### ISSUE TITLE
Restructure docs

### ISSUE THREAD

**Issue by rmainwork** [seas-computing/react-boilerplate#17]
Current boilerplate is documented, but the code is structured in such a way as to make it somewhat difficult to navigate and read. This is because Typedoc expects things to be namespaced using `Parent.Child` syntax rather than `parent/child`. Additionally, Typedoc typically documents each individual file as it's own module by default leading to messy documentation as seen below.

![image](https://user-images.githubusercontent.com/50675045/60271444-5f359280-98c0-11e9-96c1-d0411d21334c.png)

Instead, a better alternative is to add docblocks to elements where appropriate and break things up into modules - resulting in something like this:
![image](https://user-images.githubusercontent.com/50675045/60271942-3a8dea80-98c1-11e9-8f19-4a9175d7374c.png)

![image](https://user-images.githubusercontent.com/50675045/60271953-3eba0800-98c1-11e9-9538-ed42edaf9671.png)

**Comment by chrxr** [comment]
@rmainseas Can this issue be closed now?

**Comment by rmainwork** [comment]
I don't think so. The end goal of this issue is to break the docs up into smaller modules to help separate things out a big more. This is contingent upon https://github.com/christopherthielen/typedoc-plugin-external-module-name/issues/332 being fixed(which from the sounds of the comments on that issue may be the case, but it may be worth taking some time to revisit this briefly to see). The changes in #18 were just a stop-gap until the resolution of https://github.com/christopherthielen/typedoc-plugin-external-module-name/issues/332 so that the documentation was somewhat readable.
